tlair/src/test/scala/at/forsyte/apalache/tla/lir/io/TestCounterexampleWriter.scala

```diff
@@ -0,0 +1,190 @@
+package at.forsyte.apalache.tla.lir.io
+
+import java.io.{PrintWriter, StringWriter}
+
+import at.forsyte.apalache.tla.lir.convenience.tla._
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.FunSuite
+
+@RunWith(classOf[JUnitRunner])
+class TestCounterexampleWriter extends FunSuite {
+
+  def compareTla(notInvariant: NotInvariant, init: State, nextStates: List[NextState], expected: String): Unit = {
+    val stringWriter = new StringWriter()
+    val printWriter = new PrintWriter(stringWriter)
+    val writer = new TlaCounterexampleWriter(printWriter)
```

Comment:
You could extend the test suite with `BeforeAndAfterEach`. Then you could set up the repetitive constructors before each test like this:
```scala

  override def beforeEach() {
    super.beforeEach()
    stringWriter = new StringWriter()
    ...
  }
```

When the number of tests is growing, it is getting harder to maintain (and refactor!) the test initialization. So it is better to do initialization once in one place.